addons/controllers/clusterbootstrap_controller_test.go

```diff
@@ -1029,6 +1029,127 @@ var _ = Describe("ClusterBootstrap Reconciler", func() {
 			})
 		})
 	})
+
+	// This test case is for ensuring that controller is watching providerRef for core packages.
+	// 1. create antrea with foobar cr as provider, packageinstall gets generated with a given secret
+	// 2. replace secret in .status.secretRef of foobar cr provider
+	When("Cluster with external provider for CNI", func() {
+		BeforeEach(func() {
+			clusterName = "test-cluster-5"
+			clusterNamespace = "cluster-namespace-5"
+			clusterResourceFilePath = "testdata/test-cluster-bootstrap-5.yaml"
+			ns := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterNamespace,
+				},
+			}
+			err := k8sClient.Create(ctx, ns)
+			if err != nil {
+				Expect(apierrors.IsAlreadyExists(err)).To(BeTrue())
```

Comment:
Since `BeforeEach` run on every start of `It`, perhaps combine the 2 `It` into 1 and separate with `By` can avoid this duplicate creation error?

Reply:
I combined the 2 tests into one.